src/Resolvers/Microsoft.NET.Sdk.WorkloadManifestReader/WorkloadDefinition.cs

```diff
@@ -34,4 +43,14 @@ public enum WorkloadDefinitionKind
         Dev,
         Build
     }
+
+    public class WorkloadRedirect : BaseWorkloadDefinition
```

Comment:
I need more time to thing about it. So far to me, it is transparent in the workload manifest sense. However, workload ID is used many places outside manifest reader. Mostly installation record, and UI for display. As an ID, it should be unique. And now the assumption breaks. 

Every time we compare workload id, we need to consider if they are just an alias. And when we iterate workloads, which "id" we should display?

The current code should not be able to resolve installation record mismatch. We need some kind of migration of installation record. But what if there is a downgrade?

We need to re-evaluate every place we have a workloadid.toString and comparison. In general, I don't like 2 ID can point to the same thing (nuget package id being inconsistently case sensitive/ in sensitive caused a lot bugs already). It is a lot for the caller to watch out. Maybe we could override "==" and "toString". But I still prefer to just have another layer of "display id" and never change the "true id"